### PR TITLE
Sorbet: init

### PR DESCRIPTION
**Pull request by vidusheeamoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
This PR adds the `sorbet/` directory and `typed` sigil to all files in our codebase. Some tests are expected to fail ATM, I am pushing changes to fix them.

### CODE REVIEW

File: Library/Homebrew/PATH.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# typed: false
```
Comment (MikeMcQuaid) [inline Library/Homebrew/PATH.rb:1]: What decides if a file is `false`/`ignore`/`true`?

**Comment by issyl0** [comment]
> Also: is there any other way to specify typed/not than at the top of every file (e.g. a configuration file that lists all files)? Is/can there be a default e.g. default to false/true/ignore?

Sorbet supports YAML configuration for this, which seems like a good option: https://sorbet.org/docs/cli#overriding-strictness-levels

**Comment by vidusheeamoli** [comment]
> Sorbet supports YAML configuration for this, which seems like a good option: [sorbet.org/docs/cli#overriding-strictness-levels](https://sorbet.org/docs/cli#overriding-strictness-levels)

This would also not interrupt with `test/support/fixtures`, which is a good thing. I will start working on this then!

File: Library/Homebrew/PATH.rb (hunk 1 above)
Comment (vidusheeamoli) [reply Library/Homebrew/PATH.rb:1]: `ignore` does not show any errors, `false` shows syntax errors and `true` reports type errors. When `srb tc` is run for the very first time, it shows 0 errors (despite having some files as `true`).
IIUC, then `srb init` assigns each file, the highest degree of the `typed` sigil till it shows no errors initially. 
@issyl0 please confirm. Thank you! 😄

**Comment by MikeMcQuaid** [comment]
Nice work, thanks @vidusheeamoli!

File: Library/Homebrew/PATH.rb (hunk 1 above)
Comment (issyl0) [reply Library/Homebrew/PATH.rb]: Correct!

**Pull request merged**